### PR TITLE
fix observable mapping

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
In `getRunResultCiemss` we establish how PyCIEMSS result should map back to the model, our mapping logic seem to be out-of-date for observable variables, which cause the ground-truth not not appear in calibrate operation.

This PR should fix the logic, but also mark getRunResultCiemss as deprecated as we start to replace the chart.js charts with Vegalite, which generally wants a more simplified data format that is closer to the raw produced data.

image

### Testing
- Do a calibrate operation, the mapping should map an observed variable in the model to the grounding data
- When calibrate finishes, chart the observed variable, should see both the forecast-values and ground-truth-values

### CODE REVIEW

File: packages/client/hmi-client/src/services/models/simulation-service.ts [hunk 1]
```diff
@@ -147,6 +147,8 @@ export async function getCalibrateBlobURL(runId: string) {
 	return resp.data.url;
 }
 
+// @deprecated - The notion of RunResult is a outdated with introduction of Vegalite charts
+// that use a more barebone setups closwer to the raw data
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/models/simulation-service.ts:151]:
```suggestion
// that use a more barebone setup closer to the raw data
```

File: packages/client/hmi-client/src/services/models/simulation-service.ts [hunk 2]
```diff
@@ -171,20 +173,31 @@ export async function getRunResultCiemss(runId: string, filename = 'result.csv')
...
-				outputRowRunResults[keyName] = inputRow[key];
+				runConfigs[newKey].push(Number(inputRow[key]));
+				return;
+			}
+			if (key.endsWith('_state')) {
+				const newKey = key.replace(/_state$/, '');
+				outputRowRunResults[newKey] = inputRow[key];
+				if (!runConfigs[newKey]) {
+					runConfigs[newKey] = [];
+				}
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/models/simulation-service.ts:190]: Should this test be done before those `if`s?

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/services/models/simulation-service.ts (hunk 2 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/services/models/simulation-service.ts:190]: nope, each if block generates a different 'newKey'

**Review by blanchco — APPROVED** [review]
Lgtm

**Pull request merged**